### PR TITLE
Add Run Locally button in Task Inspector webpage.

### PR DESCRIPTION
**Pull request by GianaBhateja**

### CODE REVIEW

File: src/lib/ui/taskinfo.jsx [hunk 1]
```diff
@@ -254,6 +255,15 @@ const TaskInfo = React.createClass({
     window.location.href = '../task-creator';
   },
 
+  printRun() {
```
Comment (djmitche) [inline src/lib/ui/taskinfo.jsx]: There's no 'printing' in browser-side JS, so this is an odd name.  Also, I don't see the function called?

File: src/lib/ui/runLocally.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        </Modal.Header>
+        <Modal.Body>
+          <span>{this.props.children}</span>
+          {
+            this.state.executing ? (
+              <div>
+                <hr />
+                <h4>Status</h4>
+                <span>
+                  {this.renderWaitFor('result')}
```
Comment (djmitche) [inline src/lib/ui/runLocally.jsx]: I don't see `this.state.result` ever being set?

File: src/lib/ui/runLocally.jsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { Button, Glyphicon, Modal } from 'react-bootstrap';
+import * as utils from '../utils';
+
+/** Button with an associated confirm dialog */
+const RunLocally = React.createClass({
```
Comment (djmitche) [inline src/lib/ui/runLocally.jsx]: This class looks like basically a copy of `ConfirmAction` -- aside from the class name, it doesn't have anything to do with running locally..

**Comment by GianaBhateja** [comment]
Thanks dustin.I will make the changes and will complete it.

**Comment by GianaBhateja** [comment]
Hi, @imbstack @djmitche 

I just made the changes. Some feedback needed! Actually, what I need to do is to render a dialog box containing the Run Locally script on clicking "Run Locally" button present at the bottom of the webpage Task Inspector. Is this the correct way to do it or I need to do it in some other way?(actually I am beginner in ReactJS)

**Comment by GianaBhateja** [comment]
Screenshots of the webpage.
selection_027
selection_028

File: src/lib/ui/taskinfo.jsx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
-import { Table, Label } from 'react-bootstrap';
+import { Button, Table, Label } from 'react-bootstrap';
```
Comment (djmitche) [inline src/lib/ui/taskinfo.jsx]:
```
/home/dustin/p/taskcluster-tools/src/lib/ui/taskinfo.jsx
  2:10  warning  'Button' is defined but never used  no-unused-vars
```

File: src/lib/ui/taskinfo.jsx [hunk 5]
```diff
@@ -221,7 +221,6 @@ const TaskInfo = React.createClass({
   rerunTask() {
     return this.queue.rerunTask(this.props.status.taskId);
   },
-
```
Comment (djmitche) [inline src/lib/ui/taskinfo.jsx]: I expect @eliperelman will want this newline to remain

**Comment by GianaBhateja** [comment]
Ok. I will remove the lint warnings.

File: src/lib/ui/runLocally.jsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
```
Comment (eliperelman) [inline src/lib/ui/runLocally.jsx]: Please rename this file to `RunLocally.js`. For this repo I am deprecating the creation of new `.jsx` files in favor of `.js` files, and want filenames to start matching the import/export of the module. Thanks!

File: src/lib/ui/runLocally.jsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import { Button, Modal } from 'react-bootstrap';
+
+/** Button with an associated dialog containing code to run locally */
+const RunLocally = React.createClass({
+
```
Comment (eliperelman) [inline src/lib/ui/runLocally.jsx]: Remove this blank line.

File: src/lib/ui/taskinfo.jsx (hunk 5 above)
Comment (eliperelman) [inline src/lib/ui/taskinfo.jsx]: Yep! Please add this newline back in.

**Comment by GianaBhateja** [comment]
@eliperelman please let me know how should I proceed.

**Comment by djmitche** [comment]
@eliperelman let's not let the perfect be the enemy of the good - I don't think moving into the actions tab is in-scope for this PR (although, @GianaBhateja, you are more than welcome to work on that next!)

That said, I kind of like the idea of showing the run-locally text on the main page, rather than in a modal, so that it has more space.  So basically the button would reveal that text.  That seems like a smaller modification.  It's also what was [originally suggested](https://bugzilla.mozilla.org/show_bug.cgi?id=1210398#c7).

So, to proceed: make the three line-by-line modifications @eliperelman has requested, and modify the button to reveal the text on the page, rather than in a modal.  That may mean you don't need a `RunLocally.js` anymore, depending on how you structure the code.

**Comment by eliperelman** [comment]
@djmitche I think that's a good call. Let's go with that.

**Comment by GianaBhateja** [comment]
@djmitche, @eliperelman Thanks! I will make the modification as suggested.

File: src/lib/ui/taskinfo.jsx [hunk 8]
```diff
@@ -15,6 +15,16 @@ const TaskInfo = React.createClass({
...
   },
 
+  getInitialState() {
+    return {
+      scriptLoaded: false
+    };
+  },
+
+  onClick() {
+    this.setState({ scriptLoaded: true });
```
Comment (djmitche) [inline src/lib/ui/taskinfo.jsx]: Let's call this "showRunLocallyScript"

File: src/lib/ui/taskinfo.jsx [hunk 9]
```diff
@@ -15,6 +15,16 @@ const TaskInfo = React.createClass({
     task: React.PropTypes.object.isRequired
   },
 
+  getInitialState() {
+    return {
+      scriptLoaded: false
+    };
+  },
+
+  onClick() {
```
Comment (djmitche) [inline src/lib/ui/taskinfo.jsx]: Let's call this "onRunLocally"

Comment (eliperelman) [inline src/lib/ui/taskinfo.jsx]: I commonly see `handle<Action>` as commonplace in the React community, which I think makes sense too. Let's go with `handleRunLocally`.

File: src/lib/ui/taskinfo.jsx [hunk 10]
```diff
@@ -200,18 +210,24 @@ const TaskInfo = React.createClass({
...
-            </tr>
-
-            <tr>
-              <td>Run Locally</td>
-              <td>
-                <Code language="bash">
-                  {this.renderRunLocallyScript()}
-                </Code>
+                  buttonSize="small" />&nbsp;
+                <Button type="submit" bsSize="small" bsStyle="default" onClick={this.onClick }>
```
Comment (eliperelman) [inline src/lib/ui/taskinfo.jsx]: Remove extra space after `.onClick`.

File: src/lib/ui/taskinfo.jsx [hunk 11]
```diff
@@ -200,18 +210,24 @@ const TaskInfo = React.createClass({
...
+                  Run Locally
+                </Button>
               </td>
             </tr>
+              {(this.state.scriptLoaded ? (
```
Comment (eliperelman) [inline src/lib/ui/taskinfo.jsx]: Remove opening parenthesis before `this.`

File: src/lib/ui/taskinfo.jsx [hunk 12]
```diff
@@ -200,18 +210,24 @@ const TaskInfo = React.createClass({
...
+                  <td>Run Locally</td>
+                    <td>
+                      <Code language="bash">
+                        {this.renderRunLocallyScript()}
+                      </Code>
+                    </td>
+                  </tr>
+                ) :
+                null
+              )}
```
Comment (eliperelman) [inline src/lib/ui/taskinfo.jsx]: Remove extra closing parenthesis.

**Comment by GianaBhateja** [comment]
@djmitche,  @eliperelman Changes done, Please have a look.

File: src/lib/ui/taskinfo.jsx [hunk 13]
```diff
@@ -15,6 +15,16 @@ const TaskInfo = React.createClass({
...
   },
 
+  getInitialState() {
+    return {
+      showRunLocallyScript: false
+    };
+  },
+
+  handleRunLocally() {
+    this.setState({ showRunLocallyScript: true });
```
Comment (eliperelman) [inline src/lib/ui/taskinfo.jsx]: In order to make this button toggleable, let's change this to:

```js
this.setState({ showRunLocallyScript: !this.state.showRunLocallyScript });
```

Comment (GianaBhateja) [inline src/lib/ui/taskinfo.jsx]: modification done.

**Comment by eliperelman** [comment]
Great job!

**Comment by GianaBhateja** [comment]
@eliperelman  Thanks.

**Pull request merged**